layouts/community/ergodox/azurefog/keymap.c

```diff
@@ -0,0 +1,382 @@
+#include QMK_KEYBOARD_H
+
+
+enum custom_keycodes {
+  RGB_SLD= SAFE_RANGE, // can always be here
+  RGB_SLD,
+
+};
+
+
+enum {
+  TD_QUOT_SEMI = 0,
+  TD_PGUP_HOME = 1,
+  TD_PGDN_END = 2,
+  TD_LEFT_CLEFT = 3,
+  TD_RIGHT_CRIGHT = 4,
+  TD_BKSPC = 5,
+  TD_DEL_INS = 6
+};
+
+int i,j;
+
+uint8_t row_def[5][13] = {
+{0xFF, 0x00, 0x00, 0, 1, 2, 3, 4, 24, 25, 26, 27, 28},
+{0xFF, 0x80, 0x00, 5, 6, 7, 8, 9, 29, 30, 31, 32, 33},
+{0xFF, 0xFF, 0x00, 10, 11, 12, 13, 14, 34, 35, 36, 37, 38},
+{0x00, 0xFF, 0x00, 15, 16, 17, 18, 19, 39, 40, 41, 42, 43},
+{0x00, 0xFF, 0xFF, 20, 21, 22, 23, 44, 45, 46, 47, 47, 47}, //god forgive me for padding my array
+};
+
+void matrix_init_user() {
+  rgb_matrix_mode_noeeprom(RGB_MATRIX_SOLID_COLOR);
+  rgb_matrix_set_color_all(0x00, 0xFF, 0xFF);
+}
+
+LEADER_EXTERNS();
+void matrix_scan_user(void) {
+  LEADER_DICTIONARY() {
+    leading = false;
+    leader_end();
+    SEQ_ONE_KEY(KC_U) {
+      SEND_STRING("");
+    }
+    SEQ_ONE_KEY(KC_2) {
+      SEND_STRING(""SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_1) {
+      SEND_STRING(""SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_5) {
+      SEND_STRING(""SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_0) {
+      SEND_STRING("make ergodox_ez:azurefog"SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_A) {
+      SEND_STRING("azurefog@gmail.com"SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_R) {
+      SEND_STRING(SS_LGUI("r")"https://www.reddit.com"SS_TAP(X_ENTER));
+    }
+    SEQ_ONE_KEY(KC_N) {
+      SEND_STRING(SS_LCTRL(SS_LGUI(SS_TAP(X_RIGHT))));
+    }
+    SEQ_ONE_KEY(KC_B) {
+      SEND_STRING(SS_LCTRL(SS_LGUI(SS_TAP(X_LEFT))));
+    }
+    SEQ_ONE_KEY(KC_M) {
+      SEND_STRING(SS_LCTRL(SS_LGUI(SS_TAP(X_D))));
+    }
+    SEQ_ONE_KEY(KC_COMMA) {
+      SEND_STRING(SS_LCTRL(SS_LGUI(SS_TAP(X_F4))));
+    }
+    SEQ_ONE_KEY(KC_C) {
+      SEND_STRING(SS_LGUI("r")"cmd"SS_TAP(X_ENTER));
+    }
+  }
+}
+
+void leader_start(void) {
+  ergodox_led_all_on();
+}
+
+void leader_end(void) {
+  ergodox_led_all_off();
+}
+
+
+
+void del_dance (qk_tap_dance_state_t *state, void *user_data) {
+  switch (state->count) {
+    case 0 ... 2:
+      register_code(KC_DEL);
+      unregister_code(KC_DEL);
+      break;
+    case 3:
+      register_code(KC_INS);
+      unregister_code(KC_INS);
+      break;
+  }
+}
+
+void bs_dance (qk_tap_dance_state_t *state, void *user_data) {
+  switch (state->count) {
+    case 0 ... 1:
+      tap_code(KC_BSPACE);
+      break;
+    case 2:
+      register_code(KC_LSHIFT);
+      tap_code(KC_HOME);
+      unregister_code(KC_LSHIFT);
```

Comment:
You could replace these three lines with
```c
tap_code16(LSFT(KC_HOME));
```
It essentially does the same thing, except it's shorter and cleaner.